json-schemas/protocol-definition.json

```diff
@@ -4,29 +4,26 @@
   "type": "object",
   "additionalProperties": false,
   "required": [
-    "recordDefinitions",
+    "types",
     "records"
   ],
   "properties": {
-    "recordDefinitions": {
-      "type": "array",
-      "minItems": 1,
-      "items": {
-        "type": "object",
-        "required": ["id"],
-        "additionalProperties": false,
-        "properties": {
-          "id": {
-            "type": "string"
-          },
-          "schema": {
-            "type": "string"
-          },
-          "dataFormats": {
-            "type": "array",
-            "minItems": 1,
-            "items": {
+    "types": {
+      "type": "object",
+      "patternProperties": {
+        ".*": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "schema": {
               "type": "string"
+            },
+            "dataFormats": {
+              "type": "array",
+              "minItems": 1,
```

Comment:
Sanity: was it discussed that `dataFormats` is a required constraint?

Reply:
Update from discussion earlier today: we were originally planning on making it required, and now we've changed our minds to leave it optional

Reply:
Added a commit making `dataFormats` optional again and adding a test verifying it's optional